platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/admin/ConfigurationFilesPoller.java

```diff
@@ -58,145 +53,96 @@
      *
      * @param configurationDirectoryPath directory to watch for changes
      * @param fileExtension              extension of of the files to watch
-     * @param watchService               watch service to use
-     * @param executorService            executor service used to create the watch thread
      */
     public ConfigurationFilesPoller(@NotNull Path configurationDirectoryPath,
-            @NotNull String fileExtension, @NotNull WatchService watchService,
-            @NotNull ExecutorService executorService) {
+            @NotNull String fileExtension) {
         notNull(configurationDirectoryPath, "configurationDirectoryPath cannot be null");
         notNull(fileExtension, "fileExtension cannot be null");
-        notNull(watchService, "watchService cannot be null");
-        notNull(executorService, "executorService cannot be null");
 
         this.configurationDirectoryPath = configurationDirectoryPath;
-        this.watchService = watchService;
-        this.executorService = executorService;
         this.fileExtension = fileExtension;
     }
 
-    public void init() throws IOException {
-        LOGGER.debug("Starting {}...",
-                this.getClass()
-                        .getName());
-
+    public void register(@NotNull ChangeListener listener) {
```

Comment:
✏️  Just noticed that this class uses the `@NotNull` annotation. We changed our convention since we wrote this class to assume nothing should be `null` unless we use `@Nullable`. Feel free to remove those if you want.